app/components/article-content.js

```diff
@@ -388,6 +390,36 @@ export default Component.extend(
 			}
 		},
 
+		/**
+		 * Handles opening sections when click event occurs on reference
+		 *
+		 * @returns {void}
+		 */
+		handleReferences(event) {
+			if (event.target.closest('a[href^="#cite_ref-"]')) {
+				const element = event.target.closest('a[href^="#cite_ref-"]'),
+					currentCiteNote = document.getElementById(element.hash.slice(1, element.length)),
+					closest = currentCiteNote.closest('section[id*="section"]');
+
+				if (closest) {
+					let currentHeader = document.getElementById(closest.previousElementSibling.id);
+					if (!currentHeader.classList.contains('open-section')) {
+						currentHeader.classList.add('open-section');
+					}
+				}
+			}
+
+			if (event.target.closest('.reference a[href^="#cite_note-"]')) {
+				let header = this.element.querySelectorAll('ol.references');
+				if (header) {
+					header = header[0].closest('section[id*="section"]').previousElementSibling;
```

Comment:
try not to reuse variables, having two separate ones should make the code easier to understand